my_variant.h

```diff
@@ -0,0 +1,149 @@
+#include <cassert>
+#include <iostream>
+#include <algorithm>
+#include <type_traits>
+
+using std::cout;
+using std::endl;
+
+
+template <typename... types>
+struct max_size_helper { };
+
+template <typename T, typename... types>
+struct max_size_helper<T, types...> {
+	static size_t constexpr value = std::max(sizeof(T), max_size_helper<types...>::value);
+};
+
+template <> 
+struct max_size_helper<> {
+	static size_t constexpr value = 0;
+};
+
+
+template <typename... types>
+struct max_align_helper { };
+
+template <typename T, typename... types>
+struct max_align_helper<T, types...> {
+	static size_t constexpr value = std::max(alignof(T), max_align_helper<types...>::value);
+};
+
+template <> 
+struct max_align_helper<> {
+	static size_t constexpr value = 0;
+};
+
+/**
+* get index of type from parameter pack
+*/
+
+template <typename T, typename... Ts> struct get_index;
+
+template <typename T, typename... Ts>
+struct get_index<T, T, Ts...> : std::integral_constant<std::size_t, 0> {};
+
+template <typename T, typename Tail, typename... Ts>
+struct get_index<T, Tail, Ts...> :
+    std::integral_constant<std::size_t, 1 + get_index<T, Ts...>::value> {};
+
+template <typename T>
+struct get_index<T> {
+	static_assert(sizeof(T) == 0, "Element is not found");
+};
+
+template <class Obj, class... types> struct destroy_all;
+template <class Head, class... types>
+struct my_variant {
+	my_variant() : m_member(new (&m_aligned_storage) Head{}) { }
+
+	my_variant(my_variant<Head, types...>&) {
+		assert(sizeof(Head) == 0 && "Not implemented");
```

Comment:
Are you using and operator on a string? What am I missing?

Reply:
I am using a limited c assert which does not provide an interface for a message (like static_assert) and this is the patchy way to issue a message. If the lhs condition is not satisfied the rhs is ignored, otherwise the string will be part of the message. It is ugly, but in real life I will have nice library Macros/Methods to it :)